### PR TITLE
Mdx provider

### PR DESCRIPTION
**Pull request by maiieul**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests

# Description

TLDR: I think this will require manual testing :sweat_smile: 

Introducing MDXProvider to remove boilerplate from .mdx files. This should substantially reduce development and maintenance efforts.

- Moved all components into `/components` and `/examples` folders for easier retrieval
- Fixed import aliases
- Added `<MDXProvider>` to root layout with `components` object from `mdx-components.tsx`, which is possible thanks to `recma-provide-components.tsx` script (implemented through vite.config.ts) 
- added `rehypePrettyCode` plugin (also used by qwik's docs) with two small rehype scripts for Copying the Code without having to use a component in the markdown.
- Rewrote headless and fluffy routes using new `Example` and `CodeSnippet` components that take advantage of import.meta.glob
   - Accordion
   - Combobox
   - Modal
   - Popover
   - Select
   - Separator
   - Tabs
   - Toggle
   - Tooltip
- Cleaned up some code (typos, repetitive CSS classes found with the tailwind extension, typescript type annotations made into generics, non anymore needed q:slots)
   
This involved moving lots of files around but it's actually not that big of a change, yet it's not a small one either.

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-ui/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by maiieul** [comment]
I think we can reuse the `<Example name="accordion/hero />` pattern with `<APITable />`, `<KeyboardInteractionTable />`  and `<AnatomyTable />`. What do you think?

**Comment by maiieul** [comment]
Please, don't review for now. I'm trying to split this into smaller PRs.
